### PR TITLE
Sending start services log after enable

### PR DESCRIPTION
**Pull request by MatkovIvan**

### CODE REVIEW

**Comment by guperrot** [comment]
It is actually intended to send start service log even when disabled. How does iOS or UWP behave?

File: sdk/appcenter/src/main/java/com/microsoft/appcenter/AppCenter.java [hunk 1]
```diff
@@ -115,6 +115,11 @@
      */
     private Set<AppCenterService> mServices;
 
+    /**
+     * Started services for which the log isn't sent yet.
+     */
+    private List<String> mStartedServicesNamesToLog;
```
Comment (guperrot) [inline sdk/appcenter/src/main/java/com/microsoft/appcenter/AppCenter.java]: You can use Collection

File: sdk/appcenter/src/main/java/com/microsoft/appcenter/AppCenter.java [hunk 2]
```diff
@@ -655,6 +672,12 @@ private void setInstanceEnabled(boolean enabled) {
             StorageHelper.PreferencesStorage.putBoolean(PrefStorageConstants.KEY_ENABLED, true);
         }
 
+        /* Send started services. */
+        if (mStartedServicesNamesToLog != null && switchToEnabled) {
```
Comment (guperrot) [inline sdk/appcenter/src/main/java/com/microsoft/appcenter/AppCenter.java]: Looks like 1 unit test is missing for code coverage

File: sdk/appcenter/src/main/java/com/microsoft/appcenter/AppCenter.java [hunk 3]
```diff
@@ -566,12 +571,24 @@ private void finishStartServices(Iterable<AppCenterService> services) {
...
+        sendStartServiceLog(serviceNames);
+    }
 
-        /* Queue start service log. */
+    /**
+     * Queue start service log.
+     *
+     * @param serviceNames the services to send.
+     */
+    private void sendStartServiceLog(List<String> serviceNames) {
```
Comment (guperrot) [inline sdk/appcenter/src/main/java/com/microsoft/appcenter/AppCenter.java]: `@WorkerThread`

File: sdk/appcenter/src/main/java/com/microsoft/appcenter/AppCenter.java (hunk 1 above)
Comment (MatkovIvan) [reply sdk/appcenter/src/main/java/com/microsoft/appcenter/AppCenter.java]: `StartServiceLog` required `List` which in turn is required for `JSONUtils`. So, this change may affect some other code. `List` looks good for me here, let me know if you will nor change your mind.

File: sdk/appcenter/src/main/java/com/microsoft/appcenter/AppCenter.java (hunk 2 above)
Comment (MatkovIvan) [reply sdk/appcenter/src/main/java/com/microsoft/appcenter/AppCenter.java]: Not for this line, I think

**Pull request merged**